abi/README.md

```diff
@@ -0,0 +1,51 @@
+# ABI Package
+
+## Overview
+The ABI package provides functionality for marshaling and unmarshaling actions without importing the codec package. It is designed to work across different language implementations.
+
+## ABI Format
+The ABI is defined in JSON format, as shown in the `abi.json` file:
+```json
+{
+    "actions": [
+        {
+            "id": 1,
+            "action": "MockObjectSingleNumber"
+        },
+        //...
+    ],
+    "types": [
+        {
+            "name": "MockObjectSingleNumber",
+            "fields": [
+                {
+                    "name": "Field1",
+                    "type": "uint16"
+                }
+            ]
+        },
+        //...
+}
+```
+
+The ABI consists of two main sections:
+- actions: A list of action definitions, each with an id and action name.
+- types: A list of type definitions, each with a name and a list of fields.
+
+## Implementation
+To create an implementation of this package in any other langauge:
+- Copy the testdata folder.
+- Ensure all marshaling is identical to the Go implementation.
+- JSON files should align with their corresponding .hex files.
```

Comment:
Let's add a little more detail here. Could we explain the structure of the testdata folder and how an implementation should use it to ensure compliance?